contracts/events/TopicEvent.sol

```diff
@@ -442,8 +442,8 @@ contract TopicEvent is ITopicEvent, Ownable {
     {
         uint16 index = addressManager.getLastOracleFactoryIndex();
         address oracleFactory = addressManager.getOracleFactoryAddress(index);
-        address newOracle = IOracleFactory(oracleFactory).createCentralizedOracle(_centralizedOracle, address(this), 
-            name, resultNames, numOfResults, _bettingStartBlock, _bettingEndBlock, _resultSettingStartBlock, 
+        address newOracle = IOracleFactory(oracleFactory).createCentralizedOracle(address(this), 
+            numOfResults, _centralizedOracle, _bettingStartBlock, _bettingEndBlock, _resultSettingStartBlock, 
             _resultSettingEndBlock, addressManager.startingOracleThreshold());
```

Comment:
Looking at this numOfResults var, it seems that the only purpose is to sentinel check in Oracle constructor. Since we control the entrance of CO creation, we can do the check before creation outside of Oracle. It's your call.

Reply:
My point is that since resultNames are moved out of Oracle scope, its kinda strange to keep the count in it.

Reply:
We have a modifier of `isValidResult` to make sure an invalid result cannot be voted or bet on in the Oracles.